qiskit/qpy/formats.py

```diff
@@ -108,6 +128,25 @@
 CUSTOM_CIRCUIT_DEF_HEADER_PACK = "!Q"
 CUSTOM_CIRCUIT_DEF_HEADER_SIZE = struct.calcsize(CUSTOM_CIRCUIT_DEF_HEADER_PACK)
 
+# CUSTOM_CIRCUIT_INST_DEF_V2
+CUSTOM_CIRCUIT_INST_DEF_V2 = namedtuple(
+    "CUSTOM_CIRCUIT_INST_DEF",
+    [
+        "gate_name_size",
+        "type",
+        "num_qubits",
+        "num_clbits",
+        "custom_definition",
```

Comment:
Just curiosity, is it possible to express this with `custom_definition_size` of zero or non-zero rather than having explicit boolean flag along with size?